### PR TITLE
SDP Cache upgrade

### PR DESCRIPTION
**Pull request by dylanratcliffe**

### CODE REVIEW

File: engine.go [hunk 1]
```diff
@@ -459,12 +462,20 @@ func (e *Engine) Start() error {
...
 	// Start purging cache
-	e.cache.StartPurger()
+	e.cache.StartPurger(e.cacheContext)
 
 	return e.connect()
 }
 
+// prepCache Prepares the cache for use
+func (e *Engine) prepCache() {
+	e.cache = sdpcache.NewCache()
```
Comment (DavidS-om) [inline engine.go:476]:
```suggestion
	if e.cache != nil {
		e.cache = sdpcache.NewCache()
	}
```
or should this go into a NewEngine() call that pushes this code to the owner of the Engine?

**Review by DavidS-om — APPROVED** [review]
This is still waiting for https://github.com/overmindtech/sdpcache/pull/20; pre-approving anyways

File: engine.go (hunk 1 above)
Comment (dylanratcliffe) [reply engine.go:476]: I know, I thought about that too... It changes the whole way that you'd create engines which means that the sources would need to change, but it kind of needs it...

Comment (dylanratcliffe) [reply engine.go:476]: Aaargh you're right I'm gonna do it

**Comment by dylanratcliffe** [comment]
@DavidS-om I think you were right about creating `NewEngine()`. That has now been implemented if you could please re-review

**Review by DavidS-om — APPROVED** [review]
so much red :-D

**Review by DavidS-om — APPROVED** [review]
should we do this consistently across all repos?

**Comment by dylanratcliffe** [comment]
As far as I know this (and therefore also sources) are the only things that use it, so yes I'll update the sources once I've merged this and cut a release

**Pull request merged**